### PR TITLE
Add Array.resize method

### PR DESCRIPTION
**Pull request by bendmorris**
From https://github.com/HaxeFoundation/haxe/issues/6173

hxcpp PR: https://github.com/HaxeFoundation/hxcpp/pull/697

### CODE REVIEW

File: tests/unit/src/unitstd/Array.unit.hx [hunk 1]
```diff
@@ -278,4 +278,16 @@ var b : Dynamic = a.filter(function(x) return x & 1 == 0).map(function(x) return
 b.length == 2;
 b[0] == 0;
 b[1] == 20;
-#end
\ No newline at end of file
+#end
+
+// resize
+var a : Array<Int> = [1,2,3];
+a.length == 3;
```
Comment (RealyUniqueName) [inline tests/unit/src/unitstd/Array.unit.hx]: This line is unnecessary, because this is not test of array construction.

**Comment by RealyUniqueName** [comment]
Let's try to find the most effective way to resize an array for each target.
E.g. for js just changing the length of an array may be sufficient:
```javascript
var a = [1, 2, 3, 4];
a.length = 2;
console.log(a); //[1, 2]
```
https://try.haxe.org/#D1cC2

**Comment by bendmorris** [comment]
Nice, that seems to work for JS. Updated.

**Comment by RealyUniqueName** [comment]
Should also work for flash iirc.

**Comment by nadako** [comment]
This also needs `eval` implementation

**Comment by ncannasse** [comment]
HL support also requires adding resize() to ArrayBase (with a throw) , ArrayBytes, and ArrayDyn

**Comment by RealyUniqueName** [comment]
What's the status here?

**Comment by Justinfront** [comment]
For flash I would also recommend    arr.length = 0;

**Comment by bendmorris** [comment]
I don't know how to add methods to Array for cpp without modifying hxcpp. Still waiting on a response over there.

>For flash I would also recommend arr.length = 0;

That is already in this PR.

**Comment by Simn** [comment]
I have restarted the CI, it fails on flash and as3 at the moment.

**Pull request merged**